File: tests/output/jpa/src/main/javagen/restaurant/jpa_sequence_metamodel/dtos/restaurant/CommandeItem.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package restaurant.jpa_sequence_metamodel.dtos.restaurant;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import jakarta.annotation.Generated;

import restaurant.jpa_sequence_metamodel.enums.restaurant.StatutCommandeCode;

/**
 * Détail d'une commande en liste.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public interface CommandeItem {

	/**
	 * Identifiant de la commande.
	 */
	Integer getId();

	/**
	 * Date et heure de la commande.
	 */
	LocalDateTime getDateCommande();

	/**
	 * Montant total de la commande.
	 */
	BigDecimal getMontantTotal();

	/**
	 * Statut de la commande.
	 */
	StatutCommandeCode getStatutCommandeCode();

	/**
	 * Client ayant passé la commande.
	 */
	Integer getClientId();

	/**
	 * Hydrate values of instance.
	 * @param id value to set.
	 * @param dateCommande value to set.
	 * @param montantTotal value to set.
	 * @param statutCommandeCode value to set.
	 * @param clientId value to set.
	 */
	void hydrate(Integer id, LocalDateTime dateCommande, BigDecimal montantTotal, StatutCommandeCode statutCommandeCode, Integer clientId);
}

File: tests/output/focus/src/model/restaurant/commande-item.ts
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

import {e, entity, EntityToType} from "@focus4/entities";
import {DO_CODE, DO_DATE_HEURE, DO_ID, DO_ID_2, DO_PRIX} from "../../domains";

import {StatutCommande} from "./enums";

export type CommandeItem = EntityToType<CommandeItemEntityType>;
export type CommandeItemEntityType = typeof CommandeItemEntity;

export const CommandeItemEntity = entity({
    id: e.field(DO_ID_2, f => f
        .label("restaurant.commande.id")
    ),
    dateCommande: e.field(DO_DATE_HEURE, f => f
        .label("restaurant.commande.dateCommande")
    ),
    montantTotal: e.field(DO_PRIX, f => f
        .label("restaurant.commande.montantTotal")
    ),
    statutCommande: e.field(DO_CODE, f => f.type<StatutCommande>().defaultValue("EN_ATT")
        .label("restaurant.commande.statutCommande")
    ),
    clientId: e.field(DO_ID, f => f
        .label("restaurant.commande.clientId")
    )
});
